docs/source/migrating-from-gateway.mdx

```diff
@@ -36,53 +38,69 @@ The Apollo Router supports most (non-deprecated) `APOLLO_`-prefixed environment
+## Constructor options (including plugins) passed to `new ApolloServer({ ... })`
 
-In general, we're interested in minimizing the need for users to write plugins. In that spirit, prior to writing a plugin, we encourage users to open issues on our GitHub repository with use-cases so we can iterate on possible generalized solutions.
+There are a large variety of uncommon parameters that can be passed to `new ApolloServer({ ... })`, though we'll focus on the most common ones:
 
-That said, some customizations - or time constraints - will likely necessitate some users write plugins.  The Apollo Router's plugins are today written in Rust.  In the future, we expect to offer the ability to write scripted behavior in other languages.  To facilitate writing Rust plugins, we've put together a series of examples of common plugins and those can be found in the Customizations section of our docs.
+- `context`
+- `plugins`
 
-TODO
+To see the full list of options, see [`ApolloServer` options](https://www.apollographql.com/docs/apollo-server/api/apollo-server#options).  If you're using other options, additional migration steps may be necessary to replicate the same behavior.  Please open a discussion on our GitHub repository so we can better understand your needs and hopefully help you with a solution.
+
+### `context`
+
+The `context` parameter is an object that is which TODO
+
+### `plugins`
+
+It was often necessary to write various plugins for Apollo Server, which is the foundation that Apollo Gateway runs on.  These plugins are passed to the `new ApolloServer({ ... })` constructor options in the `plugins` array.  Understanding what each plugin which was enabled will be important to note.
+
+Generally speaking, Apollo Router is trying to minimize the number of cases where custom plugins are necessary to perform common behaviors, for example, setting custom HTTP status codes, transmitting telemetry data, etc.  To facilitate this, Apollo Router has a YAML-structured configuration based which enables common behaviors:
+
+- Propagating particular/all headers to subgraphs
+- OpenTelemetry instrumentation and distributed tracing support
+- Adjusting logging verbosity
+
+To try to reduce the amount of user code that lives in each Apollo Router, prior to re-implementing existing Apollo Gateway plugin behavior in a new language (Rust), we encourage our community to start (or engage on existing) conversations about desired customizations in the [GitHub Discussions on the Router repository](https://github.com/apollographql/router/discussions/).  There are various core plugins we think the Apollo Router could offer out of the box and we'd like to help discuss their design.
```

Comment:
```suggestion
Prior to re-implementing existing Apollo Gateway plugin behavior in a new language (Rust), we encourage our community to start (or engage on existing) conversations about desired customizations in the [GitHub Discussions on the Router repository](https://github.com/apollographql/router/discussions/).  There are various core plugins we think the Apollo Router could offer out of the box and we'd like help to discuss their design.
```

Reply:
I was somewhat trying to paint the picture of some of the pains of shared user code living in a centralized place and what that means to evolving a federated graph (usually someone still has to "own" that code), but agree it's better without this.  That's probably deserving of a whole article. 😉